paper/src/main/java/com/devotedmc/ExilePearl/listener/PlayerListener.java

```diff
@@ -864,27 +862,33 @@ public void onPearlThrow(ProjectileLaunchEvent e) {
 		if (!(e.getEntity() instanceof EnderPearl)) {
 			return;
 		}
-
 		final Player p = (Player)e.getEntity().getShooter();
 		if (p == null) {
 			return;
 		}
+		ItemStack mainHand = p.getInventory().getItemInMainHand();
+		ItemStack offHand = p.getInventory().getItemInOffHand();
 
-		ExilePearl pearl = pearlApi.getPearlFromItemStack(p.getInventory().getItemInMainHand());
+		ExilePearl pearl = pearlApi.getPearlFromItemStack(mainHand) == null ? pearlApi.getPearlFromItemStack(offHand) : pearlApi.getPearlFromItemStack(mainHand);
 		if (pearl == null) {
 			return;
 		}
-
+		if (pearlApi.getPearlConfig().getFreeByThrowing()) {
+			if (pearl.validateItemStack(mainHand)) {
+				mainHand.setAmount(mainHand.getAmount() - 1);
+			} else if (pearl.validateItemStack(offHand)){
+				offHand.setAmount(offHand.getAmount() - 1);
+			}
+			pearlApi.freePearl(pearl, PearlFreeReason.PEARL_THROWN);
```

Comment:
This method returns a boolean of success, I havent handled this returning false since it would probably be explicitly only ever handled by ExilePearls config

Reply:
This works for me now, thanks